firmware/PIMORONI_BADGER2040W/lib/badger2040.py

```diff
@@ -249,15 +249,17 @@ def ip_address(self):
         import network
         return network.WLAN(network.STA_IF).ifconfig()[0]
 
-    def connect(self):
+    def connect(self, **args):
         from network_manager import NetworkManager
         import WIFI_CONFIG
         import uasyncio
         import gc
 
+        status_handler = args.get("status_handler", self.status_handler)
+
         if WIFI_CONFIG.COUNTRY == "":
             raise RuntimeError("You must populate WIFI_CONFIG.py for networking.")
         self.display.set_update_speed(2)
```

Comment:
It would be good to remove this too, really it should have been in the function above. Its a strange side effect, connecting to wifi changes the speed the display updates at

Reply:
Agreed, I'll move it into the handler.